source/features/comment-shortcuts.tsx

```diff
@@ -0,0 +1,100 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const shortcutKeys = new Set(['e', 'd', 'h', 'j', 'k', '+', '-', 'n']);
+const shortcutClass: Record<string, string> = {
+	e: '.rgh-edit-comment, [aria-label="Edit comment"]',
+	d: '[aria-label="Delete comment"]',
+	h: '[aria-label="Hide comment"]',
+	'+': '[data-reaction-label="+1"]',
+	'-': '[data-reaction-label="-1"]'
+};
+
+function commentHash(comment: HTMLElement): string {
+	// What do I call this var??
+	const reviewComment = comment.closest('.js-comment[id^="pullrequestreview-"]');
+	const {hash} = reviewComment ? reviewComment.querySelector<HTMLAnchorElement>('a.js-timestamp')! : comment.querySelector<HTMLAnchorElement>('a.js-timestamp')!;
+
+	return reviewComment ? hash + '-body-html' : hash;
+}
+
+function triggerShortcut(shortcut: string) {
+	const selectedComment = select(location.hash)!;
+	select<HTMLButtonElement>(shortcutClass[shortcut], selectedComment)?.click();
+}
+
+function focusComment({delegateTarget: comment}: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	if (select.exists(':target')) {
+		console.log('I dont work correctly');
+	}
+
+	history.replaceState({}, document.title, commentHash(comment));
+}
+
+function init(): void {
+	delegate(document, '.timeline-comment.unminimized-comment, .review-comment', 'click', focusComment);
+	document.addEventListener('keypress', event => {
+		if (!shortcutKeys.has(event.key) || event.target instanceof HTMLTextAreaElement || event.target instanceof HTMLInputElement) {
+			return;
+		}
+
+		event.preventDefault();
+
+		if (['j', 'k'].includes(event.key)) {
+			const items = select.all<HTMLAnchorElement>('.timeline-comment-group[id^="issue"]:not([href]), .timeline-comment-group[id^="pullrequestreview-"]:not([href])');
+			// `j` goes to the next comment `k` goes back a comment
+			const direction = event.key === 'j' ? 1 : -1;
+			// Find current
+			const currentComment = /^#issue|^#pullrequestreview-/.test(location.hash) ? select<HTMLAnchorElement>(location.hash)! : select<HTMLAnchorElement>(':target')!;
+			const currentIndex = items.indexOf(currentComment);
+			// Nothing selected or were on the first comment
+			if (currentIndex + direction < 0 || currentIndex === -1) {
+				return;
+			}
+
+			// Find chosen
+			const chosen = items[Math.min(currentIndex + direction, items.length - 1)]; // Clamp it so it cant go past the last one
+
+			// Focus comment and dont put into history
+			location.replace(commentHash(chosen));
+			chosen.scrollIntoView({
+				block: 'start'
+			});
```

Comment:
It's the default? https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

```suggestion
			chosen.scrollIntoView();
```